Sources/Sentry/SentrySerialization.m

```diff
@@ -16,35 +16,17 @@
 @implementation SentrySerialization
 
 + (NSData *_Nullable)dataWithJSONObject:(NSDictionary *)dictionary
-                                  error:(NSError *_Nullable *_Nullable)error
 {
-// We'll do this whether we're handling an exception or library error
-#define SENTRY_HANDLE_ERROR(__sentry_error)                                                        \
-    SENTRY_LOG_ERROR(@"Invalid JSON: %@", __sentry_error);                                         \
-    *error = __sentry_error;                                                                       \
-    return nil;
-
-    NSData *data = nil;
-
-#if defined(DEBUG) || defined(TEST) || defined(TESTCI)
-    @try {
-#else
-    if ([NSJSONSerialization isValidJSONObject:dictionary]) {
-#endif
-        data = [NSJSONSerialization dataWithJSONObject:dictionary options:0 error:error];
-#if defined(DEBUG) || defined(TEST) || defined(TESTCI)
-    } @catch (NSException *exception) {
-        if (error) {
-            SENTRY_HANDLE_ERROR(NSErrorFromSentryErrorWithException(
-                kSentryErrorJsonConversionError, @"Event cannot be converted to JSON", exception));
-        }
```

Comment:
I understand this code was weird, but it did help me to diagnose some json serialization issues in the past. `isValidJSONObject` only returns `YES` or `NO` but the exception tells you what is invalid. Maybe we should consider just always doing the `@try`/`@catch` even in production and including that info in any errors we send to ourselves.

Reply:
We had to rewrite this part of the code to be able to understand it and solve a bug that affects everyone, and ultimately, the CI code was not the same as the release code, what was why we did not detect it early.

AFAIK is a good practice to avoid @try/catch for objc when possible. 
But I agree that having the actual error description would help a lot, something we dont get with `isValidJSONObject`.

@philipphofmann any thoughts?

Reply:
Agreed it's good practice to avoid @try/@catch, but it was only compiled that way for non-prod. It was only used in debug/test to help us diagnose things during development.

I'd be surprised if we ultimately were seeing different outcomes between catching an exception or looking at the inout error, vs calling isValidJSONObject, as the documentation states either method is appropriate to validate the input.

Reply:
The error was fixed not we have a better undestanding of what was happening, I dont see a problem to use the compiler declarative again, as long as we dont make a "spaghetti" code, is hard to understand. 
Is ok to duplicate code to have only one `#if/#else`, way better to read the code.

Reply:
Using compiler directives led to having untested code, which led to crashes in production. If we want extra context when debugging I would prefer using something like `options.debug` and test all the code properly. Then our users would also see the extra context when they are debugging.